bot/modules/orders/commands.js

```diff
@@ -0,0 +1,154 @@
+// @ts-check
+const logger = require('../../../logger');
+const { Community } = require('../../../models');
+const {
+  validateBuyOrder,
+  isBannedFromCommunity,
+  validateSeller,
+  validateSellOrder,
+} = require('../../validations');
+const messages = require('../../messages');
+const ordersActions = require('../../ordersActions');
+
+const Scenes = require('./scenes');
+
+exports.buyWizard = async ctx => enterWizard(ctx, ctx.user, 'buy');
+exports.sellWizard = async ctx => enterWizard(ctx, ctx.user, 'sell');
+
+exports.sell = async ctx => {
+  try {
+    const user = ctx.user;
+    // Sellers with orders in status = FIAT_SENT, have to solve the order
+    const isOnFiatSentStatus = await validateSeller(ctx, ctx, user);
+
+    if (!isOnFiatSentStatus) return;
+
+    const sellOrderParams = await validateSellOrder(ctx);
+
+    if (!sellOrderParams) return;
+    const { amount, fiatAmount, fiatCode, paymentMethod, priceMargin } =
+      sellOrderParams;
+    let communityId = null;
+    let community = null;
+    // If this message came from a group
+    // We check if the there is a community for it
+    if (ctx.message.chat.type !== 'private') {
+      // Allow find communities case insensitive
+      const regex = new RegExp(
+        ['^', '@' + ctx.message.chat.username, '$'].join(''),
+        'i'
+      );
+      community = await Community.findOne({ group: regex });
+      if (!community) {
+        ctx.deleteMessage();
+        return;
+      }
+      communityId = community._id;
+    } else if (user.default_community_id) {
+      communityId = user.default_community_id;
+      community = await Community.findOne({ _id: communityId });
+    }
+    // We verify if the user is not banned on this community
+    if (await isBannedFromCommunity(user, communityId)) {
+      await messages.bannedUserErrorMessage(ctx);
+      return;
+    }
+    // If the user is in a community, we need to check if the currency is supported
+    if (!!community && !community.currencies.includes(fiatCode)) {
+      await messages.currencyNotSupportedMessage(ctx, community.currencies);
+      return;
+    }
+    const order = await ordersActions.createOrder(ctx.i18n, ctx.bot, user, {
```

Comment:
We need to pass bot object as a parameter of this function to use it here

Reply:
ctx instead of ctx.bot should work as well. Ideally **createOrder** should not be sending messages. It should throw an exception and the caller should handle the error.

Reply:
ctx instead of ctx.bot should work as well. Ideally **createOrder** should not be sending messages. It should throw an exception and the caller should handle the error.

Reply:
Yes we should handle errors in a better way, curious I didn't know that ctx == bot, it works